janus_core/src/time.rs

```diff
@@ -32,49 +35,40 @@ impl Debug for RealClock {
     }
 }
 
-#[cfg(feature = "test-util")]
-pub mod test_util {
-    use crate::{
-        message::{Duration, Time},
-        time::Clock,
-    };
-    use std::sync::{Arc, Mutex};
-
-    /// A mock clock for use in testing. Clones are identical: all clones of a given MockClock will
-    /// be controlled by a controller retrieved from any of the clones.
-    #[derive(Clone, Debug)]
-    #[non_exhaustive]
-    pub struct MockClock {
-        /// The time that this clock will return from [`Self::now`].
-        current_time: Arc<Mutex<Time>>,
-    }
+/// A mock clock for use in testing. Clones are identical: all clones of a given MockClock will
+/// be controlled by a controller retrieved from any of the clones.
+#[derive(Clone, Debug)]
+#[non_exhaustive]
+pub struct MockClock {
```

Comment:
I think `MockClock` should remain guarded by the `test-util` feature, since it is intended for use only in tests. (What's the technical reason you'd like to drop this feature guard?)

Reply:
I need this functionality to allow submitting reports with crafted timestamps. If I were to turn on the `janus_core/test-util` feature from the `interop_binaries` dependencies, then that would result in it being enabled for all builds throughout the workspace, and we'd be building `kube` from our `janus_server` release builds, etc.

Reply:
That's unfortunate. I suppose this is another artifact of the interplay between workspaces & features/how feature unification is implemented.

I think one of the solutions at https://nickb.dev/blog/cargo-workspace-and-the-feature-unification-pitfall might help us. I think our release builds also effectively use solution #2 as well since we specify the `--bin` flag to `cargo build` (but maybe specifying `-p` is also required?). Are you sure our release builds would encounter this pitfall?

In any case, I don't think we need to hold this PR up over this issue, but I'd definitely like to resolve it somehow. Would you mind filing an issue? I'd like to be able to use features w/o dependency resolution changing what features are enabled on a per-package basis; hopefully one of the blog solutions will work, though I'd be happier if we didn't have to go as far as introducing sub-workspaces.

Reply:
I tested `cargo --bin aggregator` with and without `-p janus_server`, and the package specification is necessary to keep the feature from leaking over.